pkg/client/helper.go

```diff
@@ -240,13 +240,18 @@ func InClusterConfig() (*Config, error) {
 	if err != nil {
 		return nil, err
 	}
+	rootCA, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/ca.cert")
+	if err != nil {
+		return nil, err
+	}
 	return &Config{
 		// TODO: switch to using cluster DNS.
 		Host:        "https://" + net.JoinHostPort(os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")),
```

Comment:
Does the cert generated by salt include the kubernetes service IP as an allowed subjectAltName? If not, TLS verification will still fail here on a hostname verification error, even if the signer verification passes with the root CA bundle.